### PR TITLE
Enable PILOT_ENABLE_IP_AUTOALLOCATE by default for Istio 1.25

### PR DESCRIPTION
**Pull request by howardjohn**

### CODE REVIEW

File: releasenotes/notes/ip-allocation-v2-default.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: release-notes/v2
+kind: feature
+area: traffic-management
+issue:
+  - 53596
+releaseNotes:
+  - |
+    **Promoted** the `PILOT_ENABLE_IP_AUTOALLOCATE` value to default to `true`.
+    This enables the new iteration of [IP auto-allocation]*/latest/docs/ops/configuration/traffic-management/dns-proxy/#address-auto-allocation),
```
Comment (craigbox) [inline releasenotes/notes/ip-allocation-v2-default.yaml:9]:
```suggestion
    This enables the new iteration of [IP auto-allocation](/docs/ops/configuration/traffic-management/dns-proxy/#address-auto-allocation),
```

**Review by bleggett — APPROVED** [review]
Does this merit starting a 1.24 [compatibility profile](https://github.com/istio/istio/blob/master/manifests/helm-profiles/compatibility-version-1.23.yaml)?

**Comment by ilrudie** [comment]
> Does this merit starting a 1.24 [compatibility profile](https://github.com/istio/istio/blob/master/manifests/helm-profiles/compatibility-version-1.23.yaml)?

maybe. It could be a bit jarring if you are using the original DNS IP feature, upgrade and suddenly things work in a new way.

**Comment by bleggett** [comment]
> > Does this merit starting a 1.24 [compatibility profile](https://github.com/istio/istio/blob/master/manifests/helm-profiles/compatibility-version-1.23.yaml)?
> 
> maybe. It could be a bit jarring if you are using the original DNS IP feature, upgrade and suddenly things work in a new way.

I asked mostly because we either need to maintain those, according to some specific set of rules that's unambiguous, or we need to drop them.

**Comment by howardjohn** [comment]
Added compat version. Thanks!

**Review by jacob-delgado — APPROVED** [review]

File: tests/integration/pilot/common/routing.go [hunk 2]
```diff
@@ -3829,8 +3829,8 @@ spec:
 			statusCode: http.StatusOK,
 			from:       t.Apps.A,
 			to:         fakeExternalAddress,
-			protocol:   protocol.HTTPS,
-			port:       443,
```
Comment (hzxuzhonghu) [inline tests/integration/pilot/common/routing.go:3833]: can you exxplain why this changed

Comment (howardjohn) [reply tests/integration/pilot/common/routing.go:3833]: the test before was expecting to send to a  bogus IP with an SNI set. With auto allocation there is a VIP and Istio will configure a listener for an IP match, not an SNI match

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**